src/_json.js

```diff
@@ -16,6 +16,36 @@ function parseJSON(json) {
   return JSON.parse(json, json_parse)
 }
 
+function parseJSONStreaming(json) {
+  var results = []
+
+  try {
+    results.push(parseJSON(json))
+  } catch (error) {
+    var position
+
+    /** JSON parse error message is throw by the browser */
+    if (error.message.match(/JSON.parse/g)) {
+      var matchResult = error.message.match(/column ([0-9+]*)/)
+      position = matchResult[1] - 1
+      /** Unexpected token message is throw by NodeJS */
+    } else if (error.message.match(/Unexpected token/g)) {
+      var matchResult = error.message.match(/at position ([0-9+]*)$/)
+      position = matchResult[1]
+    } else {
+      throw error
+    }
+
+    var left = json.slice(0, position)
+    var right = json.slice(position)
```

Comment:
`slice` is for arrays. Why don't we use `substr` here? Then you don't have to do position math (above).

Reply:
I think for this case there is no diff, and String also has a `slice()` method.

https://stackoverflow.com/questions/2243824/what-is-the-difference-between-string-slice-and-string-substring